### DISCUSSION TITLE
Default volume mounts when debugging dotnet core docker images

### DISCUSSION THREAD

**Discussion by urosman** [microsoft/vscode-docker#2729] [Q&A]
Im developing a dotnet core application which is extensible through plugins. The application has it's own docker image and the application is located on /app while plugins can be mounted somewhere inside the /app folder so they can be loaded.

I am looking into plugin development and would like to use the applications docker image using the docker-run configuration but I'm running into some trouble because when I enable debugging in the docker-run configuration, the default logic for dotnet core mounts application (in my case a plugin) into /app overriding my application.

My docker-run configuration:
```
{
    "type": "docker-run",
    "label": "docker-run: debug",
    "dependsOn": [
        "docker-build: debug"
    ],
    "dockerRun": {},
    "netCore": {
        "appProject": "${workspaceFolder}/DockerHelloWorld.csproj",
        "enableDebugging": true
    }
},
```

Would it be an option to make the debugging options configurable by specifying some target path where to put the build artefacts or is there a better way for solving this?

I guess one of the options would be to set the `enableDebugging` to false and manually mount the debugger into the image.

**Comment by bwateratmsft** [comment]
Yeah, I think you would have to set `enableDebugging` to false. We have logic where if another volume is mapped to `/app`, then it will "win", but if no volume is mapped at all (which would be this case), it will apply the `/app` mapping. The most important mappings we add are [here](https://github.com/microsoft/vscode-docker/blob/main/src/tasks/netcore/NetCoreTaskHelper.ts#L234-L318):

Folder containing csproj => `/app`
Workspace folder => `/src`
`~/.vsdbg` => `/remote_debugger`
`~/.nuget/packages` => `/root/.nuget/packages`

**Comment by bwateratmsft** [comment]
Changing this over to a Discussion since it seems to fit
